packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -671,6 +794,398 @@ def to_json(self):
+                        )
+                        m_type, key = get_map_type(yaml, name)
+                        m_type = format_type(m_type)
+                        inner_model.append(
+                            Token(
+                                key + ": Map<string, " + m_type + ">;",
+                                TokenKind.Comment,
+                            )
+                        )
+                    else:
+                        self.add_new_line()
+                        self.add_whitespace(indent)
+                        m_type, key = get_map_type(yaml, name)
+                        m_type = format_type(m_type)
+                        self.add_comment(
+                            None, key + ": Map<string, " + m_type + ">;", None
+                        )
+                        self.add_new_line()
+                        self.add_whitespace(4)
+                        self.add_comment(None, "};", None)
+
+                        # START COLLECTING INNER MODEL DATA
+                        if not m_type == "{" + "}":
+                            indent = 4
+                            if "[]" in m_type:
+                                m_type = m_type[0 : len(m_type) - 2]
+                            inner_model.append(Token(" ", TokenKind.Newline))
+                            inner_model.append(
+                                Token(" " * (indent * 4), TokenKind.Whitespace)
+                            )
+
+                            inner_model.append(
+                                Token(
+                                    "model " + m_type + " {",
+                                    TokenKind.Comment,
+                                )
+                            )
+
+                else:
+                    if inner_model:
+                        inner_model.append(Token(" ", TokenKind.Newline))
+                        inner_model.append(
+                            Token(" " * (indent * 4), TokenKind.Whitespace)
+                        )
+                        if isinstance(json_request[i], list) and not isinstance(
+                            json_request[i][0], dict
+                        ):
+                            inner_model.append(Token(i, TokenKind.Comment))
+                        else:
+                            inner_model.append(Token(i + ": {", TokenKind.Comment))
+                        name = i
+                    else:
+                        self.add_new_line()
+                        if isinstance(json_request[i], list) and not isinstance(
+                            json_request[i][0], dict
+                        ):
+                            pass
+                        else:
+                            self.add_whitespace(indent)
+                            self.add_comment(None, i + ": {", None)  # + ": {"
+                        name = i
+                        inner_model = []
```

Comment:
I feel like I've seen `inner_model` defined a few times in this function. Is this to clear the list?

Reply:
yes!

Reply:
changed to clear{